File: components/credentials/src/main/java/org/cloudfoundry/credhub/views/DataResponse.java
package org.cloudfoundry.credhub.views;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.cloudfoundry.credhub.domain.CredentialVersion;

import static com.google.common.collect.Lists.newArrayList;

public class DataResponse {

  private final List<CredentialView> data;

  public DataResponse(final List<CredentialView> data) {
    super();
    this.data = data;
  }

  public static DataResponse fromEntity(final List<CredentialVersion> models) {
    final List<CredentialView> views = newArrayList();
    for (final CredentialVersion model : models) {
      if (model != null) {
        views.add(CredentialView.fromEntity(model));
      }
    }
    return new DataResponse(views);
  }

  @JsonProperty
  public List<CredentialView> getData() {
    return data;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/views/DataResponse.kt
package org.cloudfoundry.credhub.views

import com.fasterxml.jackson.annotation.JsonProperty
import com.google.common.collect.Lists
import org.cloudfoundry.credhub.domain.CredentialVersion

class DataResponse(
    @get:JsonProperty val data: List<CredentialView>,
) {
    companion object {
        fun fromEntity(models: List<CredentialVersion?>): DataResponse {
            val views: MutableList<CredentialView> = Lists.newArrayList()
            for (model in models) {
                if (model != null) {
                    views.add(CredentialView.fromEntity(model))
                }
            }
            return DataResponse(views)
        }

        fun fromEntity(
            models: List<CredentialVersion?>,
            concatenateCas: Boolean,
        ): DataResponse {
            val views: MutableList<CredentialView> = Lists.newArrayList()
            for (model in models) {
                if (model != null) {
                    views.add(CredentialView.Companion.fromEntity(model, concatenateCas))
                }
            }
            return DataResponse(views)
        }
    }
}
